src/test/java/wolox/training/controller/BookControllerTest.java

```diff
@@ -0,0 +1,191 @@
+package wolox.training.controller;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.model.Book;
+import wolox.training.repository.BookRepository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    private final List<Book> bookList = new ArrayList<>();
+
+    private String CONTENT_WITHOUT_ID;
+    private String CONTENT;
+    private Book testBook;
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private BookRepository mockBookRepository;
+
+    @BeforeEach
+    void setUp() {
+        testBook = new Book();
+        testBook.setGenre("Mystery");
```

Comment:
All these strings could be constants defined at the beginning or in another file.
This gives you the possibility that before a modification, you have to touch a couple of lines, instead of reviewing all the tests to see where it is necessary to modify